source/common/config/filter_json.cc

```diff
@@ -399,5 +399,52 @@ void FilterJson::translateClientSslAuthFilter(
                                       *proto_config.mutable_ip_white_list());
 }
 
+void FilterJson::translateGzipFilter(const Json::Object& json_config,
+                                     envoy::api::v2::filter::http::Gzip& proto_config) {
+  json_config.validateSchema(Json::Schema::GZIP_HTTP_FILTER_SCHEMA);
+
+  proto_config.set_disable_on_etag_header(json_config.getBoolean("disable_on_etag_header", false));
+
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, content_length);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, memory_level);
+  JSON_UTIL_SET_INTEGER(json_config, proto_config, window_bits);
+
+  for (const auto& json_content_type : json_config.getStringArray("content_type", true)) {
+    auto* content_type = proto_config.mutable_content_type()->Add();
+    *content_type = json_content_type;
+  }
+
+  static const NameAndEnum<envoy::api::v2::filter::http::Gzip_CompressionLevel_Enum> levels_map[] =
+      {
```

Comment:
not sure what the style guide stays, but WDYT of putting the { at the end of the previous line?

Reply:
The clang-format tool has done that.